### DISCUSSION TITLE
nv-ingest evaluation

### DISCUSSION THREAD

**Discussion by yogeshmr** [NVIDIA/nv-ingest#295] [General]
I would like to evaluate nv-ingest. I have not setup using it yet. Can someone kindly convert the data in sheet 21 and sheet 22 of the attached pdf into excel file or json file using nvidia nv-ingest and provide?
[X7RDielectric.pdf](https://github.com/user-attachments/files/18184005/X7RDielectric.pdf)

**Comment by Advait251206** [comment]
What you’re asking for (converting specific PDF sheets into Excel/JSON using **nv-ingest**) is definitely something nv-ingest *can be used for*, but the request as written has a couple of practical constraints.

---

## 1. Important limitation

nv-ingest is not a “run a PDF and directly return an Excel file” tool out of the box.

Instead, it:

- extracts text / tables / images
- produces structured intermediate JSON
- optionally sends embeddings to a vector DB

So the workflow is:

```
PDF → table extraction → structured JSON → (your code) → Excel/JSON export
```

---

## 2. About “sheet 21 and sheet 22”

This is the first ambiguity:

In most PDFs:

- “sheet” is not a real concept unless it's:
  - an embedded Excel file inside PDF
  - or a document with multiple logical sections
  - or scanned pages interpreted as table regions

So nv-ingest will interpret them as:

- page 21 / page 22
- or table regions detected on those pages

---

## 3. Correct way to do this with nv-ingest

To achieve what you want, the pipeline should look like:

### Step 1: Extract tables only

```python
Ingestor(...).files("X7RDielectric.pdf")
.extract(
    extract_text=False,
    extract_tables=True,
    extract_images=False,
)
```

---

### Step 2: Get structured JSON output

nv-ingest typically outputs something like:

```json
{
  "tables": [
    {
      "page": 21,
      "cells": [...],
      "rows": [...]
    },
    {
      "page": 22,
      "cells": [...]
    }
  ]
}
```

---

### Step 3: Convert to Excel (external step)

nv-ingest does NOT directly generate Excel.

You would do:

```python
import pandas as pd

df1 = pd.DataFrame(table_21)
df2 = pd.DataFrame(table_22)

with pd.ExcelWriter("output.xlsx") as writer:
    df1.to_excel(writer, sheet_name="page21")
    df2.to_excel(writer, sheet_name="page22")
```

---

## 4. Why the original request is tricky

The user asked:

> “convert sheet 21 and sheet 22 of PDF into excel using nv-ingest and provide”

Issues:

### ❗ nv-ingest does not:
- output Excel files directly
- guarantee “sheet-level” mapping
- isolate tables per page automatically without filtering

### ❗ You still need:
- post-processing logic
- page filtering (`page == 21 or 22`)
- DataFrame conversion

---

## 5. Recommended nv-ingest configuration

Best practice for this use case:

```python
ingestor = (
    Ingestor(...)
    .files("X7RDielectric.pdf")
    .extract(
        extract_text=False,
        extract_tables=True,
        extract_images=False,
        text_depth="page"
    )
)
```

Then filter results:

```python
results = ingestor.ingest()

tables = [r for r in results if r["page"] in [21, 22]]
```

---

## 6. If tables are not detected correctly

This is common in PDFs like dielectric/material spec sheets.

Then you need:

### Option A
```python
extract_method="nemoretriever_parse"
```

### Option B
Use OCR fallback pipeline (if scanned or semi-scanned tables)

---

## 7. Bottom line

- ✔ nv-ingest can extract the data
- ✔ you can filter pages 21 and 22
- ❌ it will NOT directly give Excel output
- ✔ you must convert JSON → Excel yourself

---

## If you want

I can show a **fully working end-to-end script** that:

- runs nv-ingest
- extracts only page 21 & 22 tables
- reconstructs clean pandas DataFrames
- exports a production-ready Excel file

---

If this answer helped or pointed you in the right direction, I'd appreciate it if you could mark it as the accepted answer so it's easier for others with the same issue to find.

Also, if you found my contribution useful, I'd appreciate it if you could check out my GitHub profile, follow me, and star any repositories you find interesting.

GitHub: https://github.com/Advait251206
